conda/recipes/ucxx/meta.yaml

```diff
@@ -40,7 +40,12 @@ requirements:
     - ninja
     - sysroot_{{ target_platform }}
   host:
+    {% if cuda_major == "11" %}
     - cudatoolkit ={{ cuda_version }}
+    {% else %}
+    - cuda-cudart-dev
```

Comment:
Can you describe how `cuda-cudart-dev` is being used? I see there isn't a CUDA compiler being installed here.

The headers in `cuda-cudart-dev` aren't usable by a host compiler (gcc) unless you have the `cuda-nvcc` compiler package installed as well. There is a circular dependency between those packages. 🙃

Reply:
> Can you describe how cuda-cudart-dev is being used? I see there isn't a CUDA compiler being installed here.

It is linking RMM, which needs thrust, so we need to pull it in here. It took me a while to figure that out too.

I guess one alternative is to pull just `cuda-cccl`, but that brings in `cuda-cccl-impl` which isn't needed, so I'm not sure what would be preferred, thus left `cuda-cudart-dev`.

> The headers in cuda-cudart-dev aren't usable by a host compiler (gcc) unless you have the cuda-nvcc compiler package installed as well. There is a circular dependency between those packages. 🙃

Assuming `cuda_runtime.h` is coming from this package (as the name seems to imply), this statement is not true. It's perfectly acceptable to call `cudaMalloc*` from host compiler.

Reply:
RMM should be fetching and patching thrust via rapids-cmake, not using the CUDA Toolkit package `cuda-cccl`.

While it is acceptable to use `<cuda_runtime.h>` from a host compiler, there is a circular dependency (currently beyond our control) between that header and the `crt/` headers shipped in `cuda-nvcc`. As a result, you actually need `cuda-nvcc` installed even if you only intend to call CUDA runtime functions from a C++ host compiler.

Reply:
I'm not sure what you're saying here exactly, are you saying we need `cuda-nvcc` installed to build? That didn't seem to be necessary, and the error was completely resolved after installing `cuda-cudart-dev`.

Reply:
🤷‍♂️ I won't argue if CI is passing. But yes, that's what I'm saying -- `cuda-nvcc` is needed to use most of `cuda-cudart-dev` even if you're using a host C++ compiler and not nvcc to build. I'm not sure what piece of `cuda-cudart-dev` is needed here, though, so maybe this is an exception.

Reply:
From what I could understand the only piece we need here is really `cuda-cccl`, but since that brought in `cuda-cccl-impl` that `cuda-cudart-dev` does not, I felt it was safer to just use `cuda-cudart-dev`. That is passing now, and to be fair I still don't get why `cuda-nvcc` would ever be required here, so since CI is passing anyway I'd prefer to keep it that way.

Reply:
Given Vyas' observation below ( https://github.com/rapidsai/ucxx/pull/61#pullrequestreview-1506222578 ), think we need the CUDA compiler

For CCCL we may want to look into using `rapids-cmake` here

In terms of `cuda-cudart-dev`, it is also needed as UCXX appears to be dynamically linking to `libcudart`, which is provided by `cuda-cudart` (pulled in and exported to packages that use `cuda-cudart-dev`)

Reply:
This is now resolved.